tests/test_elements.py

```diff
@@ -128,51 +129,73 @@ def test_ST_Equal_Column_WKTElement(self, geometry_table):
         }
 
 
-class TestWKBElement():
+class TestExtendedWKBElement():
+
+    # _bin/_hex computed by following query:
+    # SELECT ST_GeomFromEWKT('SRID=3;POINT(1 2)');
+    _bin = buffer_(b'\x01\x01\x00\x00 \x03\x00\x00\x00\x00\x00\x00'
+                   b'\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\x00@')
+    _hex = str_('010100002003000000000000000000f03f0000000000000040')
+    _srid = 3  # expected srid
+    _wkt = 'POINT (1 2)'  # expected wkt
 
     def test_desc(self):
-        e = WKBElement(b'\x01\x02', extended=True)
-        assert e.desc == '0102'
+        e = WKBElement(self._bin, extended=True)
+        assert e.desc == self._hex
 
     def test_desc_str(self):
-        e = WKBElement(str_('0102'))
-        assert e.desc == str_('0102')
+        e = WKBElement(self._hex)
+        assert e.desc == self._hex
 
     def test_function_call(self):
-        e = WKBElement(b'\x01\x02', extended=True)
+        e = WKBElement(self._bin, extended=True)
         f = e.ST_Buffer(2)
         eq_sql(f, 'ST_Buffer('
                'ST_GeomFromEWKB(:ST_GeomFromEWKB_1), '
                ':ST_Buffer_1)')
         assert f.compile().params == {
             u'ST_Buffer_1': 2,
-            u'ST_GeomFromEWKB_1': b'\x01\x02',
+            u'ST_GeomFromEWKB_1': self._bin,
         }
 
     def test_function_str(self):
-        e = WKBElement(b'\x01\x02', extended=True)
+        e = WKBElement(self._bin, extended=True)
         assert isinstance(str(e), str)
 
     def test_pickle_unpickle(self):
         import pickle
-        e = WKBElement(b'\x01\x02', srid=3, extended=True)
+        e = WKBElement(self._bin, srid=self._srid, extended=True)
         pickled = pickle.dumps(e)
         unpickled = pickle.loads(pickled)
-        assert unpickled.srid == 3
+        assert unpickled.srid == self._srid
         assert unpickled.extended is True
-        assert unpickled.data == b'\x01\x02'
+        assert unpickled.data == bytes_(self._bin)
```

Comment:
I am not sure to understand why unpickled data ara `bytes` and not `buffer`...

Reply:
I agree that relying on `buffer_` is better here.